pywinauto/linux/atspi_objects.py

```diff
@@ -0,0 +1,1225 @@
+        """
+        Try to construct RECT from _AtspiRect otherwise pass it down to RecExtMixin
+        """
+        if isinstance(otherRect_or_left, _AtspiRect):
+            self.left = otherRect_or_left.x
+            self.right = otherRect_or_left.x + otherRect_or_left.width
+            self.top = otherRect_or_left.y
+            self.bottom = otherRect_or_left.y + otherRect_or_left.height
+        else:
+            RectExtMixin.__init__(self, otherRect_or_left, top, right, bottom)
+
+
+RectExtMixin._RECT = RECT
+
+
+class _GTypeInstance(Structure):
+    pass
+
+
+class _GData(Structure):
+    pass
+
+
+class _GObject(Structure):
+    _fields_ = [
+        ('g_type_instance', _GTypeInstance),
+        ('ref_count', c_uint),
+        ('qdata', POINTER(_GData)),
+    ]
+
+
+class _GError(Structure):
+    _fields_ = [
+        ('domain', c_uint32),
+        ('code', c_int),
+        ('message', c_char_p),
+    ]
+
+
+class _GArray(Structure):
+    _fields_ = [
+        ('data', POINTER(c_char)),
+        ('len', c_uint32),
+    ]
+
+
+class _GHashTable(Structure):
+    pass
+
+
+class _GPtrArray(Structure):
+    pass
+
+
+class _GTypeInterface(Structure):
+    pass
+
+
+class _AtspiRole(Structure):
+    pass
+
+
+class _AtspiApplication(Structure):
+    pass
+
+
+class _AtspiComponent(Structure):
+    pass
+
+
+class _AtspiDocument(Structure):
+    pass
+
+
+class _AtspiImage(Structure):
+    pass
+
+
+class _AtspiStateSet(Structure):
+    pass
+
+
+class _AtspiAccessiblePrivate(Structure):
+    pass
+
+
+class _AtspiObject(Structure):
+    _fields_ = [
+        ('parent', _GObject),
+        ('accessible_parent', POINTER(_AtspiApplication)),
+        ('path', c_char_p),
+    ]
+
+
+class _AtspiAction(Structure):
+    _fields_ = [
+        ('parent', _GTypeInterface),
+    ]
+
+
+class _AtspiText(Structure):
+    _fields_ = [
+        ('parent', _GTypeInterface),
+    ]
+
+
+class _AtspiEditableText(Structure):
+    _fields_ = [
+        ('parent', _GTypeInterface),
+    ]
+
+
+class _AtspiValue(Structure):
+    _fields_ = [
+        ('parent', _GTypeInterface),
+    ]
+
+
+class _AtspiTextRange(Structure):
+    _fields_ = [
+        ('start_offset', c_int),
+        ('end_offset', c_int),
+        ('content', c_char_p),
+    ]
+
+
+class _AtspiRange(Structure):
+    _fields_ = [
+        ('start_offset', c_int),
+        ('end_offset', c_int),
+    ]
+
+
+class _AtspiAccessible(Structure):
+    pass
+
+
+_AtspiAccessible._fields_ = [
+    ('parent', _AtspiObject),
+    ('accessible_parent', POINTER(_AtspiAccessible)),
+    ('children', POINTER(_GPtrArray)),
+    ('role', _AtspiRole),
+    ('gint', c_int),
+    ('name', c_char_p),
+    ('description', c_char_p),
+    ('states', POINTER(_AtspiStateSet)),
+    ('attributes', POINTER(_GHashTable)),
+    ('cached_properties', c_uint),
+    ('priv', POINTER(_AtspiAccessiblePrivate)),
+]
+
+_AtspiStateSet._fields_ = [
+    ('fake', c_uint64 * 4),
```

Comment:
These fields can be moved to `_AtspiStateSet` declaration since there is no self-references.